src/Interceptor/Interceptor/CacheInterceptor.php

```diff
@@ -83,9 +77,18 @@ public function supportsSuffix(Instance $instance): bool
 
     public function supportsPrefix(Instance $instance): bool
     {
-        return $instance->getMethod()
-            ->hasAnnotation(Cache::class)
-        ;
+        if (!$instance->getMethod()->hasAnnotation(Cache::class)) {
+            return false;
+        }
+
+        $method = $instance->getMethod();
+        $annotation = $method->getAnnotation(Cache::class);
+
+        if ($annotation->getHandler() === 'legacy_redis') {
```

Comment:
there is legacy redis and legacy array, we should only check that it starts with legacy

Reply:
indeed, good catch